scripts/collapse_replicates.R

```diff
@@ -1,29 +1,32 @@
 suppressPackageStartupMessages(library(argparse))
 suppressPackageStartupMessages(library(magrittr))
-library(prismSeqR)
+library(prismSeqR) # Correct sourcing?
 
+# Parser object ----
 parser <- ArgumentParser()
-# specify our desired options 
 parser$add_argument("-v", "--verbose", action="store_true", default=TRUE,
                     help="Print extra output [default]")
-parser$add_argument("-q", "--quietly", action="store_false", 
-                    dest="verbose", help="Print little output")
+parser$add_argument("-q", "--quietly", action="store_false", dest="verbose", 
+                    help="Print little output")
 parser$add_argument("-c", "--lfc", default="l2fc.csv",
                     help="path to file containing l2fc values")
+parser$add_argument("--sig_cols", default="cell_set,treatment,dose,dose_unit,day", 
+                    help = "columns used to identify a unique condition")
+parser$add_argument("--cell_line_cols", default="project_code,DepMap_ID,CCLE_name", 
+                    help = "Columns that can describe a cell line")
 parser$add_argument("-o", "--out", default=getwd(), help = "Output path. Default is working directory")
 
 # get command line options, if help option encountered print help and exit
 args <- parser$parse_args()
 
-lfc_values = read.csv(args$lfc)
+# Collapse biological replicates ----
+lfc_values= data.table::fread(args$lfc, header=T, sep=',', data.table=F)
+sig_cols= unlist(strsplit(args$sig_cols, ","))
+cell_line_cols= unlist(strsplit(args$cell_line_cols, ","))
 
-print("collapsing counts")
-collapsed_counts = collapse_counts(lfc_values)
+print("Collapsing biological replicates ...")
+collapsed_l2fc= collapse_bio_reps(l2fc= lfc_values, sig_cols= sig_cols, cell_line_cols= cell_line_cols)
 
-collapsed_count_out_file = paste(
-  args$out,
-  "collapsed_values.csv",
-  sep='/'
-)
-
-write.csv(collapsed_counts, collapsed_count_out_file, row.names=F, quote=F)
+# Write out file ----
+collapsed_l2fc_path= paste(args$out, "collapsed_l2fc.csv", sep='/')
```

Comment:
@maha-naim can you update the adapter script alongside this since this file renaming will likely impact the adapter script too.